packages/pwa-kit-create-app/scripts/trim-extensions.js

```diff
@@ -319,8 +342,11 @@ function removeUnusedComponents(directory) {
                                 // Go up one level so that we can compare the parent directory path
                                 absoluteImportPath = path.resolve(path.dirname(absoluteImportPath))
                             }
-                            // If this import matches any exported file, remove it from the set
-                            if (exportedFiles.has(absoluteImportPath)) {
+                            // If this import matches any exported file and it's not from one of the component candidates, remove it from the set
+                            const isCandidate = Array.from(removeComponentCandidates).find(
+                                (candidate) => candidate.includes(path.dirname(filePath))
+                            )
```

Comment:
exportedFiles is the complete list of all exports in the code base - if we found out that the export is being imported somewhere, we don't consider it a candidate for removal, unless the import is also from a "removal candidate" directory, which means the import will also get removed. This means if we have /store-locator/compA and /store-locator/compB, and compA <-> compB, we will still be able to detect that both A and B are part of a detached graph and they can be safely removed.

Reply:
Oh good catch. I think the match needs to be === instead of includes